pkg/apis/nsx.vmware.com/v1alpha1/subnetport_types.go

```diff
@@ -19,20 +19,36 @@ type SubnetPortSpec struct {
 type SubnetPortStatus struct {
 	// Conditions describes current state of SubnetPort.
 	Conditions []Condition `json:"conditions,omitempty"`
-	// VIFID describes the attachment VIF ID owned by the SubnetPort in NSX-T.
-	VIFID string `json:"vifID,omitempty"`
-	// IPAddresses describes the IP addresses of the SubnetPort.
-	IPAddresses []SubnetPortIPAddress `json:"ipAddresses,omitempty"`
-	// MACAddress describes the MAC address of the SubnetPort.
-	MACAddress string `json:"macAddress,omitempty"`
+	// Subnet port attachment state.
+	Attachment SegmentPortAttachmentState `json:"attachment,omitempty"`
+	// Subnet gateway.
+	Gateway string `json:"gateway,omitempty"`
```

Comment:
Gateway and Netmask are not in NSX API, but we added them because VM Operator needs them?

If so, could VM Operator reads them from Subnet.Status? If these fields are not in NSX SubnetPort state, but we we have to add them in Subnet.Status, then our changes to use RealizedBindings make little sense, not better than the original `SubnetPortIPAddress`

Reply:
A reminder for this question

Reply:
> Gateway and Netmask are not in NSX API, but we added them because VM Operator needs them?

Yes.

> If so, could VM Operator reads them from Subnet.Status?

The logic may be complex. If the subnetport doesn't explicitly has spec.subnet or spec.subnetset, it will be under the default subnetset, the workflow to find the subnet is as follows:
![get_default_subnetset](https://github.com/vmware-tanzu/nsx-operator/assets/13518085/9c308560-a27c-4aef-8732-4004dc00d756)

This logic has many dependencies, if the related dependencies change in the future, it will become difficult to sync the logic change in different components. So we prefer to keep it only in NSX operator scope.

BWT, the `logicalSwitchID` is also not in the subnetport's properties, it's from the `realization_id` in API `https://{{mgr-ip}}/policy/api/v1/orgs/{{org-id}}/projects/{{project-id}}/vpcs/{{vpc-id}}/subnets/{{subnet-id}}`

Reply:
I am concerned on the Gateway and Netmask fields. Besides inconsistence with NSX APIs, they are not quite extensible. For example, how we support IPv6/dual stack, or even multiple IPs.

In NSX API, does Subnet or Subnet State include gateway and network information?

Reply:
> In NSX API, does Subnet or Subnet State include gateway and network information?

The NSX API `https://{{mgr-ip}}/policy/api/v1/orgs/{{org-id}}/projects/{{project-id}}/vpcs/{{vpc-id}}/subnets/{{subnet-id}}` has the following info:

```
{
    "ipv4_subnet_size": 64,
    "access_mode": "Private",
    "dhcp_config": {
        "enable_dhcp": false
    },
    "advanced_config": {
        "static_ip_allocation": {
            "enabled": true
        }
    },
    "resource_type": "VpcSubnet",
    "id": "sub_5b6e74b9-e262-4d51-a272-a0aff284dc88_d9813e4b-7a6d-4456-a739-15b50df79c4e",
    "display_name": "sub-domain-c10:93b3f368-e6f8-47ef--ed9dd12a-d9813e4b-7a6d-4456-a739-15b50df79c4e",
    "tags": [...],
    "path": "/orgs/default/projects/project-quality/vpcs/1bfcbf5e-a471-4265-b4c1-5a5caa4c2865/subnets/sub_5b6e74b9-e262-4d51-a272-a0aff284dc88_d9813e4b-7a6d-4456-a739-15b50df79c4e",
    "relative_path": "sub_5b6e74b9-e262-4d51-a272-a0aff284dc88_d9813e4b-7a6d-4456-a739-15b50df79c4e",
    "parent_path": "/orgs/default/projects/project-quality/vpcs/1bfcbf5e-a471-4265-b4c1-5a5caa4c2865",
    "remote_path": "",
    "unique_id": "8d628b91-cdd7-4930-a60c-90d7673ec38d",
    "realization_id": "8d628b91-cdd7-4930-a60c-90d7673ec38d",
    "owner_id": "ee431750-fcef-46a6-8767-ba7d556ac53b",
    "marked_for_delete": false,
    "overridden": false,
    "_create_time": ...,
    ...
}
```

The return of API `https://{{mgr-ip}}/policy/api/v1/orgs/{{org-id}}/projects/{{project-id}}/vpcs/{{vpc-id}}/subnets/{{subnet-id}}/status` looks like:

```
{
    "results": [
        {
            "static_ip_pool_path": "/orgs/default/projects/project-quality/vpcs/1bfcbf5e-a471-4265-b4c1-5a5caa4c2865/subnets/sub_5b6e74b9-e262-4d51-a272-a0aff284dc88_d9813e4b-7a6d-4456-a739-15b50df79c4e/ip-pools/static-ipv4-default",
            "static_ip_pool_range": "172.26.0.67-172.26.0.126",
            "gateway_address": "172.26.0.65/26",
            "network_address": "172.26.0.64/26",
            "ip_address_type": "IPV4"
        }
    ]
}
```

Reply:
Seems indeed not very convenient for VM Operator to gather all these information. Then probably let us define a single struct for VM Operator to consume.

```
NetworkInterfaceConfig {
    LogicalSwitchUUID string
    IPAddresses []NetworkInterfaceIPAddress
    MACAddress string
}

NetworkInterfaceIPAddress {
    // IP address string with the prefix
    IPAddress string
    Gateway string
}
```

Thoughts?

Reply:
Then the status will look like:
```
  status:
    attachment:
      id: 32656264-3632-4737-ad38-3266662d3432
    conditions: ...
    networkInterfaceConfig:
      logicalSwitchUUID: 2ccec3b9-7546-4fd2-812a-1e3a4afd7acc
      ipAddresses:
      - ipAddress: 172.26.0.69
        gateway: 172.26.0.65/26
      macAddress: 04:50:56:00:d0:00
```
, right?

LGTM. @dantingl @lxiaopei how do you think?